src/core_ocean/analysis_members/mpas_ocn_eliassen_palm.F

```diff
@@ -358,93 +339,49 @@ subroutine ocn_compute_eliassen_palm(domain, timeLevel, err)!{{{
-      real(KIND=RKIND), dimension(:,:), pointer :: uPVMidBuoyCoorEA
-      real(KIND=RKIND), dimension(:,:), pointer :: vPVMidBuoyCoorEA
-      real(KIND=RKIND), dimension(:,:,:), pointer :: PVFluxTest
+      integer, dimension(:), allocatable :: firstLayerBuoyCoor
+      integer, dimension(:), allocatable :: lastLayerBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: heightMidBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: heightTopBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: heightInterfaceBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: sigma
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotNormalGradOnEdge
+      real(KIND=RKIND), dimension(:,:), allocatable :: uMidBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: vMidBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: densityMidBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: densityTopBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: buoyancyMask
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotGradX
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotGradY
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotGradZ
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotGradZonal
+      real(KIND=RKIND), dimension(:,:), allocatable :: montgPotGradMerid
+      real(KIND=RKIND), dimension(:,:), allocatable :: wrk3DnVertLevelsP1
+      real(KIND=RKIND), dimension(:,:), allocatable :: wrk3DnVertLevels
+      real(KIND=RKIND), dimension(:,:), allocatable :: wrk3DBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: ErtelPVNormalGradOnEdge
+      real(KIND=RKIND), dimension(:,:), allocatable :: ErtelPVGradX
+      real(KIND=RKIND), dimension(:,:), allocatable :: ErtelPVGradY
+      real(KIND=RKIND), dimension(:,:), allocatable :: ErtelPVGradZ
+      real(KIND=RKIND), dimension(:,:,:), allocatable :: wrkVector
+      real(KIND=RKIND), dimension(:,:,:,:), allocatable :: wrkTensor
+
+      real(KIND=RKIND), dimension(:,:), allocatable :: array1_3D
+      real(KIND=RKIND), dimension(:,:), allocatable :: array2_3D
+      real(KIND=RKIND), dimension(:,:), allocatable :: array3_3D
+      real(KIND=RKIND), dimension(:,:), allocatable :: array1_3Dbuoy
+      real(KIND=RKIND), dimension(:,:), allocatable :: array2_3Dbuoy
+      real(KIND=RKIND), dimension(:,:), allocatable :: PVMidBuoyCoor
+      real(KIND=RKIND), dimension(:,:), allocatable :: PVMidBuoyCoorEA
+      real(KIND=RKIND), dimension(:,:), allocatable :: uMidBuoyCoorEA
+      real(KIND=RKIND), dimension(:,:), allocatable :: vMidBuoyCoorEA
+      real(KIND=RKIND), dimension(:,:), allocatable :: uPVMidBuoyCoorEA
+      real(KIND=RKIND), dimension(:,:), allocatable :: vPVMidBuoyCoorEA
+      real(KIND=RKIND), dimension(:,:,:), allocatable :: PVFluxTest
```

Comment:
The allocatable statements might actually need to be module-level instead of subroutine-level.  This is a non-issue once #513 is merged, but having the global `!$omp parallel` implementation could cause the subroutine-level allocatable definitions to result in segfaults if running on multiple threads.